service/controller/resource/clusterconfigmap/desired.go

```diff
@@ -134,8 +144,9 @@ func (r *StateGetter) GetDesiredState(ctx context.Context, obj interface{}) ([]*
 				"ingressController": map[string]interface{}{
 					// Legacy flag is set to true so resources created by
 					// legacy provider operators are not created.
-					"legacy": true,
+					"legacy": ingressControllerLegacy,
 				},
+				"provider": r.provider,
```

Comment:
@whites11 @rossf7 @corest  cluster-operator 023.9 with this PR hasn't been used in any platform release yet, and it's good, as it broke nginx i.e. has to be shipped together with matching nginx changes.

We're preparing cluster-operator 0.23.10 and matching nginx and azure-operator, which move nginx LB Service ownership from azure-operator to nginx IC App.

Pinged you as I'm curious what you think - I'm having doubts should setting `provider` in ingress-controller-values ConfigMap (introduced in not yet used 0.23.9) remain or not. Setting provider is not done for cluster-operator master, so it not consistent. It's also not done in cluster-values ConfigMap for all the other apps. So far there is a process and expectation that `provider` is set in every catalog ConfigMap e.g. https://github.com/giantswarm/installations/blob/master/godsmack/appcatalog/default-appcatalog-values.yaml#L6

Options are:
- keep setting provider in cluster-operator legacy
  - then we should maybe set it in cluster values ConfigMap too
  - and also introduce the same change in cluster-operator master
  - then we could also simplify new installation / catalog usage process, and cleanup existing provider configuration from installation specific catalog definition overrides
- rely on provider being set in catalog definition
  - then maybe we should remove provider being set in cluster-operator legacy for ingress-controller-values ConfigMap (i.e. stop setting it in 0.23.10, no migration needed since 0.23.9 which started emitting provider, is not used anywhere)

WDYT?

Reply:
Option 2 seems more reasonable to me. If we never used that field let's just remove it.

Reply:
OK, will remove it from 0.23.10 and release for Azure 11.4.0 and others

Reply:
Btw, we use provider info in nginx app, but that info was already passed through default app catalog ConfigMap - cluster-operator setting it in ingress-values was duplicate

Reply:
Thanks Stevo, I agree option 2 is better.

If 0.23.9 isn't used anywhere. Can we delete it from the app collections? We're still tight on resources in some CPs.

Reply:
Checked and found 0.23.9 is used in couple of aws and kvm releases which are not archived yet. Those are not affected by Azure specific changes made here so didn't need matching nginx changes.

Will ship releases in coming days with cluster-operator upgraded to 0.23.10, so eventually 0.23.9 will be removable.